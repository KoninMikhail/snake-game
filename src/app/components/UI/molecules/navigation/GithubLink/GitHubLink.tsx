import {PrimaryBorder} from '@ui/atoms/assets/borders/PrimaryBorder/PrimaryBorder';
import {ExtraShadow} from '@ui/atoms/assets/shadows/blocks/ExtraShadow/ExtraShadow';
import {StyledGithubLink} from './style/StyledGithubLink';
import {GitHubIcon} from '@ui/atoms/assets/icons/GithubIcon/GitHubIcon';
import {ExternalLink} from '@ui/atoms/navigation/ExternalLink/ExternalLink';
import {TextLink} from '@ui/atoms/typography/link/TextLink';
import {IGitHubLinkProps} from '@ui/molecules/navigation/GithubLink/GitHubLink.interface';

/**
 * GitHubLink
 *
 * Use it for GitHub link state page
 *
 * @param uri - uri to your repository
 * @param label - label for button
 */
export const GitHubLink = ({ uri, label }: IGitHubLinkProps) => (
    <ExternalLink href={uri}>
        <ExtraShadow>
            <PrimaryBorder>
                <StyledGithubLink>
                    <TextLink>{label}</TextLink>
                    <GitHubIcon size={'lg'} />
                </StyledGithubLink>
            </PrimaryBorder>
        </ExtraShadow>
    </ExternalLink>
);
