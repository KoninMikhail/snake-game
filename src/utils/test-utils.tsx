import React, {PropsWithChildren} from 'react';
import type {RenderOptions} from '@testing-library/react';
import {render} from '@testing-library/react';
import type {PreloadedState} from '@reduxjs/toolkit';
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';

import type {AppStore, RootState} from '@store/store';
import appSlice from '@store/slices/app.slice';
import soundSlice from '@store/slices/sound.slice';
import maxScoreSlice from '@store/slices/maxScore.slice';
import gameDataSlice from '@store/slices/game.slice';
import {ImagesData} from '@data/imagesData';
import {mockImagesData, mockSoundData} from '@/setupTests';
import SoundsProvider from '@context/SoundsContext';
import ImagesProvider from '@context/ImagesContext';
import {SoundsData} from '@data/soundsData';
import {DefaultTheme, ThemeProvider} from 'styled-components';
import {theme as appTheme} from '@styles/theme';

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
    preloadedState?: PreloadedState<RootState>;
    store?: AppStore;
    images?: ImagesData;
    sounds?: SoundsData;
    theme?: DefaultTheme;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        // Automatically create a store instance if no store was passed in
        store = configureStore({
            reducer: {
                app: appSlice,
                sounds: soundSlice,
                maxScore: maxScoreSlice,
                game: gameDataSlice,
            },
            preloadedState,
        }),
        images = mockImagesData,
        sounds = mockSoundData,
        theme = appTheme,
        ...renderOptions
    }: ExtendedRenderOptions = {}
) {
    function Wrapper({children}: PropsWithChildren): JSX.Element {
        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    <SoundsProvider sounds={sounds}>
                        <ImagesProvider images={images}>{children}</ImagesProvider>
                    </SoundsProvider>
                </ThemeProvider>
            </Provider>
        );
    }

    // Return an object with the store and all of RTL's query functions
    return {store, ...render(ui, {wrapper: Wrapper, ...renderOptions})};
}
