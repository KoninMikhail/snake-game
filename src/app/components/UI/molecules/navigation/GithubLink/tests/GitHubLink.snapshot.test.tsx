import { GitHubLink } from '../GitHubLink';
import { renderWithProviders } from '@utils/test-utils';

/**
 * Test
 */
describe('Github link', () => {
    const uri = 'https://github.com';
    const label = 'welcome to github';

    it('render correctly', () => {
        const { container } = renderWithProviders(<GitHubLink uri={uri} label={label} />);
        expect(container).toMatchSnapshot();
    });
});
