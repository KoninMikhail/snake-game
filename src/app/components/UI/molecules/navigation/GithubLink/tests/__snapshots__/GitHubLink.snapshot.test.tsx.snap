// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Github link render correctly 1`] = `
.c1 {
  box-shadow: rgba(0,0,0,0.25) 0px 54px 55px,rgba(0,0,0,0.12) 0px -12px 30px, rgba(0,0,0,0.12) 0px 4px 6px,rgba(0,0,0,0.17) 0px 12px 13px, rgba(0,0,0,0.09) 0px -3px 5px;
}

.c2 {
  display: grid;
  grid-template-columns: auto max-content;
  gap: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0.8rem 0.8rem 0.8rem;
  background: black;
}

.c0 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: white;
}

.c0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c3 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: white;
  font-family: 'pixeloid_bold','Segoe UI','Roboto','Oxygen','Ubuntu','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-weight: normal;
}

.c3:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media (min-width:1200px) {
  .c2 {
    padding: 0.8vw 0.8vw 0.8vw;
  }
}

<div>
  <a
    class="c0"
    href="https://github.com"
    rel="noopener noreferrer"
    target="_blank"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        style="border: 0.25em dashed deeppink;"
      >
        <span
          class="c3"
        >
          welcome to github
        </span>
        <div
          fill="currentColor"
          height="1.8em"
          width="1.8em"
        />
      </div>
    </div>
  </a>
</div>
`;
